From 82e4d5241cee41a7fed1d10707f6d8df2bc41b76 Mon Sep 17 00:00:00 2001
From: Vivek Goyal <vgoyal@redhat.com>
Date: Tue, 19 Aug 2014 22:02:49 -0400
Subject: [crypto] pefile: Handle pesign using the wrong OID

Message-id: <1408485785-777-18-git-send-email-vgoyal@redhat.com>
Patchwork-id: 88374
O-Subject: [RHEL7.1 PATCH 17/33] pefile: Handle pesign using the wrong OID
Bugzilla: 1129779
RH-Acked-by: David Howells <dhowells@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

BZ:1129779

commit dd7d66f21b9eb6a3979d8c9ba910eba772cfbbc9
Author: Vivek Goyal <vgoyal@redhat.com>
Date: Tue, 8 Jul 2014 18:10:46 +0100

The pesign utility had a bug where it was using OID_msIndividualSPKeyPurpose
instead of OID_msPeImageDataObjId - so allow both OIDs.

Signed-off-by: Vivek Goyal <vgoyal@redhat.com>
Acked-by: Vivek Goyal <vgoyal@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/crypto/asymmetric_keys/mscode_parser.c b/crypto/asymmetric_keys/mscode_parser.c
index 09336c3..214a992 100644
--- a/crypto/asymmetric_keys/mscode_parser.c
+++ b/crypto/asymmetric_keys/mscode_parser.c
@@ -58,7 +58,13 @@ int mscode_note_content_type(void *context, size_t hdrlen,
   return -EBADMSG;
  }
 
- if (oid != OID_msIndividualSPKeyPurpose) {
+ /*
+  * pesign utility had a bug where it was putting
+  * OID_msIndividualSPKeyPurpose instead of OID_msPeImageDataObjId
+  * So allow both OIDs.
+  */
+ if (oid != OID_msPeImageDataObjId &&
+     oid != OID_msIndividualSPKeyPurpose) {
   pr_err("Unexpected content type OID %u\n", oid);
   return -EBADMSG;
  }
-- 
1.7.1